### DISCUSSION TITLE
how to set multiple share ref usage

### DISCUSSION THREAD

**Discussion by nani1008** [react-hook-form/react-hook-form#4868] [Q&A] [answered]
How can I use multiple share ref usage https://react-hook-form.com/faqs#Howtosharerefusage like this one?

**Comment by nani1008** [comment, marked as answer]
const { ref: ref1, ...code1 } = register("code1", { required: true });
  const { ref: ref2, ...code2 } = register("code2", { required: true });

solve this by, is this the correct way? @bluebill1049

**Reply by bluebill1049** [reply]
That's correct for now, we are working on a potential better syntax.
